src/cursor.rs

```diff
@@ -1,247 +1,82 @@
-    pub fn move_up(&mut self, row_count: u16) -> Result<&mut TerminalCursor> {
-        self.cursor.move_up(row_count)?;
-        Ok(self)
-    }
-
-    /// Moves the cursor `col_count` times right.
-    pub fn move_right(&mut self, col_count: u16) -> Result<&mut TerminalCursor> {
-        self.cursor.move_right(col_count)?;
-        Ok(self)
-    }
-
-    /// Moves the cursor `row_count` times down.
-    pub fn move_down(&mut self, row_count: u16) -> Result<&mut TerminalCursor> {
-        self.cursor.move_down(row_count)?;
-        Ok(self)
-    }
+use crate::utils::Result;
 
-    /// Moves the cursor `col_count` times left.
-    pub fn move_left(&mut self, col_count: u16) -> Result<&mut TerminalCursor> {
-        self.cursor.move_left(col_count)?;
-        Ok(self)
-    }
+use crate::utils::Command;
 
-    /// Saves the cursor position.
-    ///
-    /// See the [restore_position](struct.TerminalCursor.html#method.restore_position) method.
-    ///
-    /// # Notes
-    ///
-    /// The cursor position is stored globally and is not related to the current/any
-    /// `TerminalCursor` instance.
-    pub fn save_position(&self) -> Result<()> {
-        self.cursor.save_position()
-    }
-
-    /// Restores the saved cursor position.
-    ///
-    /// See the [save_position](struct.TerminalCursor.html#method.save_position) method.
-    pub fn restore_position(&self) -> Result<()> {
-        self.cursor.restore_position()
-    }
+mod ansi;
+pub(crate) mod sys;
 
-    /// Hides the cursor.
-    ///
-    /// See the [show](struct.TerminalCursor.html#method.show) method.
-    pub fn hide(&self) -> Result<()> {
-        self.cursor.hide()
-    }
-
-    /// Shows the cursor.
-    ///
-    /// See the [hide](struct.TerminalCursor.html#method.hide) method.
-    pub fn show(&self) -> Result<()> {
-        self.cursor.show()
-    }
-
-    /// Enables or disables the cursor blinking.
-    ///
-    /// # Notes
-    ///
-    /// Windows versions lower than Windows 10 do not support this functionality.
-    pub fn blink(&self, blink: bool) -> Result<()> {
-        self.cursor.blink(blink)
-    }
-}
-
-/// Creates a new `TerminalCursor`.
-///
-/// # Examples
-///
-/// Basic usage:
-///
-/// ```no_run
-/// use crossterm::{cursor, Result};
-///
-/// fn main() -> Result<()> {
-///     let cursor = cursor();
-///     cursor.save_position()?;
-///
-///     cursor.goto(10, 10)?;
-///     cursor.blink(true)?;
-///
-///     cursor.blink(false)?;
-///     cursor.restore_position()
-/// }
-/// ```
-pub fn cursor() -> TerminalCursor {
-    TerminalCursor::new()
-}
-
-/// A command to move the cursor to the given position.
+/// A command that moves the terminal cursor to the given position (column, row).
 ///
 /// # Notes
-///
-/// Commands must be executed/queued for execution otherwise they do nothing.
-pub struct Goto(pub u16, pub u16);
+/// - The counting of the given dimensions starts from 0 were column 0 and row 0 is the top left.
```

Comment:
* Empty line after the heading.
* Just a nitpick, we do use `*` for the unnumbered lists in the code.
* There's typo `were` -> `where`

Can we make it shorter?

* The counting of the given dimensions starts from 0 were column 0 and row 0 is the top left.

... to ...

* Top left cell is represented as `0, 0`.